### PR TITLE
Set Openbsd malloc flags and other bits.

### PR DESCRIPTION
**Pull request by vext01**
Hi,

As mentioned yesterday in-person, in hindsight this would be a series of smaller PRs. Sorry about that.

 * Set OpenBSD's `MALLOC_OPTIONS` to sane "as deterministic as possible" values.
 * Add a sanity check for the above.
 * Unify all sanity checks so they all call a since function `spawn_sanity_check()`.
 * Rename "misc sanity checks" to "platform sanity checks.
 * Add more tests to make sections of code I touched more robust.
 * Attach log file earlier.
 * Improve logging messages.

Clearly some squashing can be done.

OK?

### CODE REVIEW

File: krun/platform.py [hunk 1]
```diff
@@ -250,6 +258,14 @@ class UnixLikePlatform(BasePlatform):
     FORCE_LIBRARY_PATH_ENV_NAME = "LD_LIBRARY_PATH"
     REBOOT = "reboot"
 
+    def bench_env_changes(self):
+        # Force libkruntime into linker path.
+        # We are working on the assumption that no-one else uses
+        # LD_LIBRARY_PATH (or equivalent) elsewhere. EnvChangeSet will check
+        # this and crash out if this assumption is invalid.
+        change = EnvChangeSet(self.FORCE_LIBRARY_PATH_ENV_NAME, LIBKRUNTIME_DIR)
+        return [change]
```
Comment (ltratt) [inline krun/platform.py]: It would actually be shorter to make line 266 `return [...]` :)

File: krun/platform.py [hunk 2]
```diff
@@ -7,17 +7,20 @@
...
 from abc import ABCMeta, abstractmethod, abstractproperty
-from krun.env import EnvChangeSet
+from krun.env import EnvChangeSet, EnvChange
 
 NICE_PRIORITY = -20
 BENCHMARK_USER = "krun"  # user is expected to have made this
 DIR = os.path.abspath(os.path.dirname(__file__))
 LIBKRUNTIME_DIR = os.path.join(DIR, "..", "libkruntime")
 
+
```
Comment (vext01) [inline krun/platform.py]: woops

File: krun/tests/__init__.py [hunk 3]
```diff
@@ -4,6 +4,18 @@
 import pytest
 
 
+def subst_env_arg(lst, var):
+    """E.g. replace list elements like 'MYVAR=something' with 'MYVAR='"""
```
Comment (ltratt) [inline krun/tests/__init__.py]: This isn't replacing things -- it's making a copy.

File: krun/tests/test_linuxplatform.py [hunk 4]
```diff
@@ -86,3 +86,23 @@ def test_tickless0004(self, monkeypatch, platform):
...
+                  'env', 'LD_LIBRARY_PATH=']
+
+        platform.isolated_cpu = 3
+
+        args = subst_env_arg(platform.bench_cmdline_adjust([], {}), "LD_LIBRARY_PATH")
+        assert args == expect
+
+    def test_bench_cmdline_adjust0002(self, platform):
+        expect = ['sudo', '-u', 'krun', 'nice', '-20', 'taskset', '0x8',
+                       'env', 'MYENV=some_value', 'LD_LIBRARY_PATH=', 'myarg']
```
Comment (ltratt) [inline krun/tests/test_linuxplatform.py]: Broken indent.

File: krun/vm_defs.py [hunk 5]
```diff
@@ -447,10 +444,14 @@ def __init__(self, vm_path):
...
-    def run_exec(self, entry_point, benchmark, iterations, param, heap_lim_k):
+    def run_exec(self, entry_point, benchmark, iterations, param, heap_lim_k,
+                 force_dir=None):
         # Duplicates generic implementation. Need to pass args differently.
 
-        script_path = os.path.join(BENCHMARKS_DIR, benchmark, entry_point.subdir, entry_point.target)
+        script_path = self._get_benchmark_path(benchmark, entry_point,
+                                               force_dir=force_dir)
+
+        # Note the double minus before arguments.
```
Comment (ltratt) [inline krun/vm_defs.py]: I noted it. But why should I note it?

**Comment by ltratt** [comment]
I made a few minor(ish) comments, nothing too scary.

**Comment by vext01** [comment]
Fixed!

**Comment by vext01** [comment]
Ah one more tweak coming.

**Comment by vext01** [comment]
Tweak applied. OK?

**Comment by ltratt** [comment]
Looks good. Squash away and then I'll merge.

**Comment by vext01** [comment]
ready.

**Comment by ltratt** [comment]
Thanks!

**Pull request merged**